### DISCUSSION TITLE
ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects

### DISCUSSION THREAD

**Discussion by Karl-Philippe** [Project-MONAI/MONAILabel#1128] [Q&A]
**Bug description**
Cannot run download monailabel due to this Error. I have properly download numpy with pip.

**To Reproduce**
pip install monailabel

**Expected behavior**

**Environment**
OS: Ubuntu 22.04
python 3.10.6
numpy 1.23.4

**Additional context**
Collecting monailabel
  Using cached monailabel-0.5.2-202210242052-py3-none-any.whl (9.3 MB)
Collecting python-multipart==0.0.5
  Using cached python_multipart-0.0.5-py3-none-any.whl
Collecting torch>=1.7
  Using cached torch-1.13.0-cp310-cp310-manylinux1_x86_64.whl (890.1 MB)
Collecting pynetdicom==2.0.2
  Using cached pynetdicom-2.0.2-py3-none-any.whl (1.6 MB)
Collecting PyYAML==6.0
  Using cached PyYAML-6.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (682 kB)
Collecting python-dotenv==0.20.0
  Using cached python_dotenv-0.20.0-py3-none-any.whl (17 kB)
Collecting monai[fire,gdown,ignite,itk,lmdb,nibabel,openslide,pillow,psutil,skimage,tensorboard,torchvision,tqdm]>=1.0.1
  Using cached monai-1.0.1-202210241233-py3-none-any.whl (1.1 MB)
Collecting ninja==1.10.2.3
  Using cached ninja-1.10.2.3-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (108 kB)
Collecting fastapi==0.78.0
  Using cached fastapi-0.78.0-py3-none-any.whl (54 kB)
Collecting requests-toolbelt==0.9.1
  Using cached requests_toolbelt-0.9.1-py2.py3-none-any.whl (54 kB)
Collecting bcrypt==3.2.2
  Using cached bcrypt-3.2.2-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (62 kB)
Collecting einops>=0.3.2
  Using cached einops-0.5.0-py3-none-any.whl (36 kB)
Collecting filelock==3.7.1
  Using cached filelock-3.7.1-py3-none-any.whl (10 kB)
Collecting numpymaxflow==0.0.2
  Using cached numpymaxflow-0.0.2.tar.gz (13 kB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error

  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [845 lines of output]
      Collecting setuptools
        Using cached setuptools-65.5.1-py3-none-any.whl (1.2 MB)
      Collecting numpy==1.19.2
        Using cached numpy-1.19.2.zip (7.3 MB)
        Installing build dependencies: started
        Installing build dependencies: finished with status 'done'
        Getting requirements to build wheel: started
        Getting requirements to build wheel: finished with status 'done'
        Preparing metadata (pyproject.toml): started
        Preparing metadata (pyproject.toml): finished with status 'done'
      Collecting wheel
        Using cached wheel-0.38.3-py3-none-any.whl (36 kB)
      Building wheels for collected packages: numpy
        Building wheel for numpy (pyproject.toml): started
        Building wheel for numpy (pyproject.toml): finished with status 'error'
        error: subprocess-exited-with-error
      
        × Building wheel for numpy (pyproject.toml) did not run successfully.
        │ exit code: 1
        ╰─> [819 lines of output]
        
        .................................................................
        .................................................................
        .................................................................
        .................................................................
        
        note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for numpy
      Failed to build numpy
      ERROR: Could not build wheels for numpy, which is required to install pyproject.toml-based projects
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by SachidanandAlle** [comment]
Please use python 3.9 or below.. next version of monai will support python 3.10 and so monailabel
